src/hclust.jl

```diff
@@ -691,6 +801,13 @@ function hclust(d::AbstractMatrix; linkage::Symbol = :single,
         throw(ArgumentError("Unsupported cluster linkage $linkage"))
     end
 
+    if leaforder == :barjoseph
```

Comment:
Let's also support `:optimal` as an alias (please update the dockstring):
```suggestion
    if leaforder == :barjoseph || leaforder == :optimal
```

It could be that we'll find the "more optimal" algorithm, then we'll be update to change what `:optimal` does while keeping `:barjoseph`.